### PR TITLE
CORTX-33670 : Create a Pre-merge job for DTM all2all ST for Motr

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- Task is to Create a Pre-merge job for DTM all2all ST for Motr component  and for hare component use main branch and hare repo as  https://github.com/Seagate/cortx-hare. During a motr PR raise this Jenkins job should get triggered and pass successfully for the PR to get merged. 

# Design
 We need checkout to Motr PR branch and Hare main branch under the checkout stage and under DTM-Integration-Test stage it will execute this job https://eos-jenkins.colo.seagate.com/job/Motr/job/DTM-Integration-Test/

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing : https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/Motr_Pre_Merge_DTM_Integration/40/ >> remote Repo
- https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/Motr_Pre_Merge_DTM_Integration/41/ >> local repo

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        skipDefaultCheckout()
+        timeout(time: 180, unit: 'MINUTES')
+        timestamps()
+        disableConcurrentBuilds()
+        ansiColor('xterm')  
+    }
+
+    parameters {  
+	    string(name: 'MOTR_REPO', defaultValue: 'https://github.com/Seagate/cortx-motr', description: 'Repo for Motr')
+        string(name: 'MOTR_BRANCH', defaultValue: 'main', description: 'Branch for Motr')
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:20]: please correct indentation

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+    environment {
+
+        // Motr Repo Info
+
+        GPR_REPO = "https://github.com/${ghprbGhRepository}"
+        MOTR_URL = "${ghprbGhRepository != null ? GPR_REPO : MOTR_URL}"
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:28]: rename both `MOTR_URL` variables to `MOTR_REPO`

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        MOTR_GPR_REFSEPEC = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
+        MOTR_BRANCH_REFSEPEC = "+refs/heads/*:refs/remotes/origin/*"
+        MOTR_PR_REFSEPEC = "${ghprbPullId != null ? MOTR_GPR_REFSEPEC : MOTR_BRANCH_REFSEPEC}"
+
+        //////////////////////////////// BUILD VARS //////////////////////////////////////////////////
+        // OS_VERSION and COMPONENTS_BRANCH are manually created parameters in jenkins job.
+        
+        COMPONENT_NAME = "motr".trim()
+        BRANCH = "${ghprbTargetBranch != null ? ghprbTargetBranch : COMPONENTS_BRANCH}"
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:39]: Remove `BRANCH` variable as it is not getting used

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        HARE_BRANCH = "main"
+        HARE_REPO = "https://github.com/Seagate/cortx-hare"
+    }
+    stages {
+
+        // Build motr fromm PR source code
+        stage('Chekout') {
+            steps {
+				script { build_stage = env.STAGE_NAME }
+                script { manager.addHtmlBadge("&emsp;<b>Target Branch : ${BRANCH}</b>&emsp;<br />") }
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:49]: remove this line

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                 sh """
+                    set +x
+                    echo "--------------BUILD PARAMETERS -------------------"
+                    echo "MOTR_REPO              = ${MOTR_REPO}"
+                    echo "MOTR_BRANCH           = ${MOTR_BRANCH}"
+                    echo "MOTR_PR_REFSEPEC       = ${MOTR_PR_REFSEPEC}"
+                    echo "-----------------------------------------------------------"
+                """
+                 
+                dir("motr") {
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:60]: Remove checkout code as you are triggering other job for actual execution of test

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+                    checkout([$class: 'GitSCM', branches: [[name: "${MOTR_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog'], [$class: 'SubmoduleOption', disableSubmodules: false, parentCredentials: true, recursiveSubmodules: true, reference: '', trackingSubmodules: false]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${MOTR_REPO}",  name: 'origin', refspec: "${MOTR_PR_REFSEPEC}"]]])
+
+                }
+                dir ('hare') {
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:65]: Remove checkout code as you are triggering other job for actual execution of test

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        
+        COMPONENT_NAME = "motr".trim()
+        BRANCH = "${ghprbTargetBranch != null ? ghprbTargetBranch : COMPONENTS_BRANCH}"
+        HARE_BRANCH = "main"
+        HARE_REPO = "https://github.com/Seagate/cortx-hare"
+    }
+    stages {
+
+        // Build motr fromm PR source code
+        stage('Chekout') {
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:46]: Rename stage to `Build Variables Info`

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        }
+
+        // Run DTM-Integration-Test
+        stage ("DTM-Integration Test") {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
+                    try {
+                        def buildCortxAllImage = build job: '/Motr/DTM-Integration-Test', wait: true,
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:79]: change variable `buildCortxAllImage` to `dtmTest`

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 1 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:20]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 3 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 4 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 8 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 6 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:62]: Hi @gauravchaudhari02 . i have also noticed that. its getting tagged to to some merged PR. Don't we need this step in the final job ?

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 9]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env groovy
+// CLEANUP REQUIRED
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:2]: Can we remove this comment if not required

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+                    echo "--------------BUILD PARAMETERS -------------------"
+                    echo "MOTR_REPO              = ${MOTR_REPO}"
+                    echo "MOTR_BRANCH           = ${MOTR_BRANCH}"
+                    echo "MOTR_PR_REFSEPEC       = ${MOTR_PR_REFSEPEC}"
+                    echo "-----------------------------------------------------------"
+                """
+                 
+                dir("motr") {
+
+                    checkout([$class: 'GitSCM', branches: [[name: "${MOTR_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog'], [$class: 'SubmoduleOption', disableSubmodules: false, parentCredentials: true, recursiveSubmodules: true, reference: '', trackingSubmodules: false]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${MOTR_REPO}",  name: 'origin', refspec: "${MOTR_PR_REFSEPEC}"]]])
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:59]: Remove recursive clone as all submodules are removed from Motr now. `recursiveSubmodules: true`

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 10 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 9 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy]: Done

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 6 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:60]: No. we do not need. remove it

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy (hunk 1 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:18]: still not indented correctly

File: jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        COMPONENT_NAME = "motr".trim()
+        HARE_BRANCH = "main"
+        HARE_REPO = "https://github.com/Seagate/cortx-hare"
+    }
+    stages {
+
+        // Build motr fromm PR source code
+        stage('Build Variables Info') {
+            steps {
+				script { build_stage = env.STAGE_NAME }
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/component_test/Motr_Pre_merge_DTM_Integration.groovy:45]: indent correctly

**Review by shailesh-vaidya — APPROVED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**